plugins/out_s3/s3.c

```diff
@@ -1744,9 +1795,8 @@ static void s3_upload_queue(struct flb_config *config, void *out_context)
 
             /* If retry limit was reached, discard file and remove file from queue */
             upload_contents->retry_counter++;
-            if (upload_contents->retry_counter >= MAX_UPLOAD_ERRORS) {
-                flb_plg_warn(ctx->ins, "Chunk file failed to send %d times, will not "
-                             "retry", upload_contents->retry_counter);
+            if (upload_contents->retry_counter > ctx->ins->retry_limit) {
+                s3_retry_warn(ctx, (char *) upload_contents->tag, upload_contents->m_upload_file->input_name, upload_contents->upload_time, FLB_FALSE);
```

Comment:
upload_contents (upload_queue) -> tag should already be a char * - https://github.com/fluent/fluent-bit/blob/00371e91c335d3567a6a7f5672f2fc78904ddd6b/plugins/out_s3/s3.h#L64
is this conversion necessary?